.travis.yml

```diff
@@ -1,24 +1,62 @@
-sudo: required
-
 language: python
 
 cache:
     directories:
         - $HOME/.cache/pip
 
-matrix:
+os: linux
+
+jobs:
```

Comment:
I've recently changed this in borg 1.1-maint branch too, see there:

https://github.com/borgbackup/borg/blob/1.1-maint/.travis.yml#L9

Ideas:
- run the flake8 job on a newer python, so we don't have to change that part anytime soon again
- distribute the py versions over the respective dists that have that python version native as their system python or are somehow close. that reduces jobs from N*M to N.
- test on python 3.9 also (at the time of that change, "3.9" was not working yet, thus I used "3.9-dev")
- we do not need the macOS jobs for here, they are an annoyance anyway due to frequent failures.